File: tool/check_doc_consistency.dart
import 'dart:io';

void main() {
  final root = Directory.current;
  const required = [
    'PRODUCT.md',
    'USER_FLOWS.md',
    'DESIGN.md',
    'ARCHITECTURE.md',
    'AI_SYSTEM.md',
    'DATA_SOURCES.md',
    'RULES.md',
    'docs/TRACEABILITY.md',
  ];
  final errors = <String>[];
  final documents = <String, String>{};
  for (final relative in required) {
    final file = File('${root.path}/$relative');
    if (!file.existsSync()) {
      errors.add('Eksik belge: $relative');
      continue;
    }
    final content = file.readAsStringSync();
    documents[relative] = content;
    if (!content.contains('İBB Kent Takip')) {
      errors.add('Ürün adı eksik: $relative');
    }
  }

  final product = documents['PRODUCT.md'] ?? '';
  final ai = documents['AI_SYSTEM.md'] ?? '';
  final rules = documents['RULES.md'] ?? '';
  for (final phrase in [
    'İstanbul Senin/153',
    'UrbanIncident',
    'Fotoğrafsız devam',
    'vatandaş güven skoru yoktur',
  ]) {
    if (!product.toLowerCase().contains(phrase.toLowerCase())) {
      errors.add('PRODUCT kararı eksik: $phrase');
    }
  }
  if (!ai.contains('yalnız üç dar yetenek')) {
    errors.add('AI üç yeteneğe daraltılmamış.');
  }
  if (!rules.contains('Doğrudan aktif dosya overwrite yasaktır')) {
    errors.add('RULES atomik persistence kararı eksik.');
  }

  final adrDirectory = Directory('${root.path}/docs/decisions');
  final adrCount = adrDirectory
      .listSync()
      .whereType<File>()
      .where((file) => file.path.split('/').last.startsWith('ADR-'))
      .length;
  if (adrCount < 8) {
    errors.add('En az 8 ADR gerekli; bulunan: $adrCount.');
  }

  if (errors.isNotEmpty) {
    stderr.writeln(errors.join('\n'));
    exitCode = 1;
    return;
  }
  stdout.writeln('Belge tutarlılığı doğrulandı: ${required.length} kanonik belge, $adrCount ADR.');
}

File: tool/check_doc_consistency.py
from pathlib import Path
import sys
import unicodedata

ROOT = Path(__file__).resolve().parents[1]
REQUIRED = [
    "PRODUCT.md", "USER_FLOWS.md", "DESIGN.md", "ARCHITECTURE.md",
    "AI_SYSTEM.md", "DATA_SOURCES.md", "RULES.md", "docs/TRACEABILITY.md",
]
errors = []
for path in ROOT.rglob("*.md"):
    raw = path.read_bytes()
    relative = str(path.relative_to(ROOT))
    if not raw.startswith(b"\xef\xbb\xbf"):
        errors.append(f"UTF-8 BOM eksik: {relative}")
    text = raw.decode("utf-8-sig")
    if unicodedata.normalize("NFC", text) != text:
        errors.append(f"NFC değil: {relative}")

for relative in REQUIRED:
    path = ROOT / relative
    if not path.exists():
        errors.append(f"Eksik belge: {relative}")
        continue
    text = path.read_text(encoding="utf-8-sig")
    if "İBB Kent Takip" not in text:
        errors.append(f"Ürün adı eksik: {relative}")

product = (ROOT / "PRODUCT.md").read_text(encoding="utf-8-sig")
for phrase in ["İstanbul Senin/153", "UrbanIncident", "Fotoğrafsız devam", "vatandaş güven skoru yoktur"]:
    if phrase.casefold() not in product.casefold():
        errors.append(f"PRODUCT kararı eksik: {phrase}")

adr_count = len(list((ROOT / "docs/decisions").glob("ADR-*.md")))
if adr_count < 8:
    errors.append(f"ADR sayısı yetersiz: {adr_count}")

if errors:
    print("\n".join(errors), file=sys.stderr)
    raise SystemExit(1)
print(f"Belge tutarlılığı doğrulandı: {len(REQUIRED)} belge, {adr_count} ADR")
